### PR TITLE
Modify the console.log function to truncate large files during logging

### PR DESCRIPTION
**Pull request by vivianzzhu91**
# Why

Modified the console.log function to truncate large files when doing console.log() 
Resolve the issue: https://github.com/expo/expo/issues/4340

# How

Try to change the return results in the packages/expo/src/logs to find where the data are sent to Expo Cli and check for the string length to truncate it.

# Test Plan

I tried to load a local json file with 22 MB and it worked to print 10000 characters (around 10kb) instead of giving a PayloadTooLargeError.

### CODE REVIEW

File: ios/Exponent.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2223,7 +2223,7 @@
 				TargetAttributes = {
 					78CEE2BF1ACD07D70095B124 = {
 						CreatedOnToolsVersion = 6.2;
-						DevelopmentTeam = C8D8QTF339;
+						DevelopmentTeam = 74633SH798;
```
Comment (ide) [inline ios/Exponent.xcodeproj/project.pbxproj]: Revert all changes to the iOS project.

File: apps/sandbox/app.json [hunk 2]
```diff
@@ -29,4 +29,4 @@
       "config": "metro.config.js"
     }
   }
-}
+}
```
Comment (ide) [inline apps/sandbox/app.json]: Revert

File: apps/sandbox/package.json [hunk 3]
```diff
@@ -19,4 +19,4 @@
     "expo-yarn-workspaces": "^1.0.0"
   },
   "private": true
-}
+}
```
Comment (ide) [inline apps/sandbox/package.json]: Revert

File: packages/expo/src/logs/LogSerialization.ts [hunk 4]
```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
     if (typeof item === 'string') {
       return item;
     } else {
-      return prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      //check the size of string returned
+      let result = prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      if(result.length > 10000){
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]: Set up your editor to run Prettier and run it over this file

File: packages/expo/src/logs/LogSerialization.ts [hunk 5]
```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
...
+        //truncate the result to 10kb
+        let newResult = result.substring(0,10000);
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]:
```suggestion
        let truncatedResult = result.substring(0,10000);
```

File: packages/expo/src/logs/LogSerialization.ts [hunk 6]
```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
...
+        newResult += '...';
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]: Communicate clearly what is happening with the truncation when people see this in their logs:
```suggestion
        newResult += `...(truncated to the first ${LOG_MESSAGE_MAX_LENGTH} characters)`;
```

File: packages/expo/src/logs/LogSerialization.ts [hunk 7]
```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
     if (typeof item === 'string') {
       return item;
     } else {
-      return prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      //check the size of string returned
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]: style: space between `//` and comment

```suggestion
      // check the size of string returned
```

File: packages/expo/src/logs/LogSerialization.ts [hunk 8]
```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
...
+      let result = prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      if(result.length > 10000){
+        //truncate the result to 10kb
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]:
```suggestion
        // truncate the result to 10kb
```

File: ios/Exponent.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ide) [inline ios/Exponent.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Revert

File: ios/Exponent/Supporting/Info.plist [hunk 10]
```diff
@@ -57,7 +57,7 @@
 	<key>Fabric</key>
 	<dict>
 		<key>APIKey</key>
-		<string>a25fafe9f9edee11a9882b32e0cd7a26df6e2c42</string>
+		<string>81130e95ea13cd7ed9a4f455e96214902c721c99</string>
```
Comment (ide) [inline ios/Exponent/Supporting/Info.plist]: Revert

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo/src/logs/LogSerialization.ts [hunk 11]
```diff
@@ -73,7 +73,16 @@ function _stringifyLogData(data: unknown[]): string[] {
     if (typeof item === 'string') {
       return item;
     } else {
-      return prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      let result = prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      // check the size of string returned
+      if (result.length > 10000) {
+        let truncatedResult = result.substring(0, 10000);
+        // truncate the result string to the first 10000 characters
+        truncatedResult += '...';
```
Comment (ide) [inline packages/expo/src/logs/LogSerialization.ts]: add the message explaining what is going on: https://github.com/expo/expo/pull/4419#discussion_r290526911

**Review by ide — CHANGES_REQUESTED** [review]
It looks like several file permissions changed as part of your commit (644 -> 755) -- please revert these.

**Review by esamelson — COMMENTED** [review]
looks great -- thanks for fixing this!! could you add an entry to CHANGELOG.md in the repo root? then will be good to go ahead and merge.

**Pull request merged**